batch2/batch/batch.py

```diff
@@ -13,6 +14,18 @@
 log = logging.getLogger('batch')
 
 
+class KubernetesResourceError(Exception):
+    def __init__(self, resource_type, name, namespace):
+        super().__init__()
+        self.resource_type = resource_type
+        self.name = name
+        self.namespace = namespace
+
+    def __str__(self):
+        return f"could not get {self.resource_type} with name '{self.name}' in " \
```

Comment:
Name is probably the most important here, I'd probably rephrase as "could not get resource {namespace}/{name} of kind {kind}".